plugins/shuffle/round.py

```diff
@@ -227,41 +239,45 @@ def process_shuffling(self):
         4. If player is last player he goes to the next phase ("Broadcasts Output") and broadcast the outputs.
         """
         phase = self.messages.phases[self.phase]
-        accused = None
+        accused, sender = None, None
         if self.me == self.last_player():
             sender = self.players[self.previous_player(player=self.last_player())]
```

Comment:
I know you didn't write this but this and the similar statement in the `else:` should be exactly equivalent. If not, it is surprising logic.

```python
if self.me == self.last_player():
    sender = self.players[self.previous_player(player=self.last_player())]
else:
    sender = self.players[self.previous_player()]
```

Reply:
Yah I saw that too.. it looked on the surface like the else and the if are the same exact thing.   This is written in the most annoying way possible. :/